tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1541,7 +1546,7 @@ def _gif():
       channels_msg = 'Channels must be in (None, 0, 3) when decoding GIF images'
       assert_channels = control_flow_ops.Assert(good_channels, [channels_msg])
       with ops.control_dependencies([assert_channels]):
-        return gen_image_ops.decode_gif(contents)
+        return convert_image_dtype(gen_image_ops.decode_gif(contents), dtype)
```

Comment:
We can probably remove `convert_image_dtype` calls from each of these helper functions and instead do it once at the end so that this function will now support types other than `uint8` and `uint16` for all image formats.